code/android-core-library/src/main/java/com/adobe/marketing/mobile/util/UrlUtils.java

```diff
@@ -49,4 +51,39 @@ public static boolean isValidUrl(final String stringUrl) {
     public static String urlEncode(final String unencodedString) {
         return UrlEncoder.urlEncode(unencodedString);
     }
+
+    /**
+     * Extra URL parameters and return as a {@code Map}
+     *
+     * @param queryString the {@code String} of the URL parameter section
+     * @return a {@code Map} of URL parameters
+     */
+    public static Map<String, String> extractQueryParameters(final String queryString) {
```

Comment:
Can you add some tests?

Reply:
It's ported from the 1.x utility class, and I didn't see any tests for it. Can we add tests in another PR to unlock the messaging migration work?